lib/mix/tasks/compile/live_view.ex

```diff
@@ -0,0 +1,170 @@
+
+  defp read_manifest do
+    case File.read(manifest()) do
+      {:ok, contents} -> :erlang.binary_to_term(contents)
+      _ -> {:unknown, nil}
+    end
+  end
+
+  defp write_manifest!(diagnostics) do
+    File.write!(manifest(), :erlang.term_to_binary({@manifest_version, diagnostics}))
+  end
+
+  defp manifest_older? do
+    if File.exists?(manifest()) do
+      other_manifests = Mix.Tasks.Compile.Elixir.manifests()
+      manifest_mtime = mtime(manifest())
+      Enum.any?(other_manifests, fn m -> mtime(m) > manifest_mtime end)
+    else
+      true
+    end
+  end
+
+  @doc false
+  def validate_components_calls(modules) do
+    for module <- modules,
+        Code.ensure_loaded?(module),
+        function_exported?(module, :__components_calls__, 0),
+        %{component: {mod, fun}, attrs: attrs, file: file, line: line} <- module.__components_calls__(),
+        attrs_defs = callee_attrs_defs(mod, fun) do
+      {dyn_attrs, static_attrs} = Enum.split_with(attrs, &match?({:root, _, _}, &1))
+      meta = %{file: file, line: line, callee: "#{inspect(mod)}.#{fun}/1"}
+
+      [
+        maybe_validate_required_attrs(static_attrs, dyn_attrs, attrs_defs, meta),
+        validate_undefined_attrs(static_attrs, attrs_defs, meta)
+      ]
+    end
+    |> List.flatten()
+  end
+
+  defp maybe_validate_required_attrs(static_attrs, _dyn_attrs = [], attrs_defs, meta) do
+    validate_required_attrs(static_attrs, attrs_defs, meta)
+  end
+
+  defp maybe_validate_required_attrs(_, _, _, _) do
+    []
+  end
+
+  defp validate_required_attrs(static_attrs, attrs_defs, meta) do
+    %{callee: callee, file: file, line: line} = meta
+    passed_attrs = Enum.map(static_attrs, &elem(&1, 0))
+
+    for %{name: name, opts: opts} <- attrs_defs, opts[:required], "#{name}" not in passed_attrs do
+      message = "missing required attribute `#{name}` for component `#{callee}`"
+      error(message, file, line)
+    end
+  end
+
+  defp validate_undefined_attrs(static_attrs, attrs_defs, meta) do
+    %{callee: callee, file: file} = meta
+    defined_attrs = Enum.map(attrs_defs, fn %{name: name} -> "#{name}" end)
+
+    for {name, _value, %{line: line}} <- static_attrs, name not in defined_attrs do
+      message = "undefined attribute `#{name}` for component `#{callee}`"
+      error(message, file, line)
+    end
+  end
+
+  defp callee_attrs_defs(module, fun) do
+    module.__components__()[fun]
+  end
+
+  defp project_modules do
+    files =
+      Mix.Project.compile_path()
+      |> File.ls!
+      |> Enum.sort()
+
+    for file <- files, [basename, ""] <- [:binary.split(file, ".beam")] do
+      String.to_atom(basename)
+    end
+  end
+
+  defp print_diagnostics(diagnostics) do
+    for %Diagnostic{file: file, position: line, message: message} <- diagnostics do
+      rel_file = file |> Path.relative_to_cwd() |> to_charlist()
+      IO.warn(message, [{nil, :__FILE__, 1, [file: rel_file, line: line]}])
```

Comment:
I will add a new API. For now, we can leave a TODO:

```suggestion
      # Use IO.warn(message, file: ..., line: ...) on Elixir v1.14+ 
      IO.warn(message, [{nil, :__FILE__, 1, [file: rel_file, line: line]}])
```